SparkPost/2web2ui
# FE-1145 - Add default Stylelint rules recommended by the Styled Components team

[FE-1145](https://jira.int.messagesystems.com/browse/FE-1145)

### What Changed
- Installs Stylelint packages according to Styled Components docs: https://styled-components.com/docs/tooling#stylelint
- Adjusts pre-commit hook to run Stylelint against JS files as well as CSS/SCSS files
- Removes use of `--fix` flag that is incompatible with the styled components processor

### How To Test
1. Open any component and add a new styled component that breaks a rule. For example, using a vendor prefix:
```js
import styled from 'styled-components';

const FancyComponent = styled.button`
  -webkit-appearance: none;
`
```
2. Stage the file and attempt to commit it - verify an error occurs
3. Remove the error and attempt to commit some other change in the file and verify no Stylelint error occurs

### To Do
